tests/test_nodes_utils.nim

```diff
@@ -56,4 +56,129 @@ proc testConsumeCommonPrefix*(left: seq[int]; right: seq[int];
+  }
+
+test "binary trie node parsing":
+  var x = 0
+  for c in parseNodeData:
+    let input = toBytesRange(c[0])
+    let node = c[1]
+    let kind = TrieNodeKind(node[0])
+    let one = toBytesRange(node[1])
+    let two = toBytesRange(node[2])
+    try:
+      let res = parseNode(input)
+      doAssert(kind == res.kind)
+      case res.kind
+      of KV_TYPE:
+        check(res.keyPath == one)
+        check(res.child == two)
+      of BRANCH_TYPE:
+        check(res.leftChild == one)
+        check(res.rightChild == two)
+      of LEAF_TYPE:
+        check(res.value == two)
+    except InvalidNode as E:
+      discard
+    except:
+      echo getCurrentExceptionMsg()
+      doAssert(false)
+    inc x
+
+const
+  kvData = [
+    ("\x00", "\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", "\x00\x10\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),
+    (""    , "\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", None),
+    ("\x00", "\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", None),
+    ("\x01", "\x00\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", None),
+    ("\x02", "", None)
+  ]
+
+test "binary trie kv node encoding":
+  for c in kvData:
+    let keyPath = b(c[0])
+    let node    = b(c[1])
+    let output  = b(c[2])
+
+    try:
+      check output == encodeKVNode(keyPath, node)
+    except ValidationError as E:
+      discard
+    except:
+      check(getCurrentExceptionMsg() == "len(childNodeHash) == 32 ")
+
+const
+  branchData = [
+    ("\xc8\x9e\xfd\xaaT\xc0\xf2\x0cz\xdfa(\x82\xdf\tP\xf5\xa9Qc~\x03\x07\xcd\xcbLg/)\x8b\x8b\xc6", "\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p",
+       "\x01\xc8\x9e\xfd\xaaT\xc0\xf2\x0cz\xdfa(\x82\xdf\tP\xf5\xa9Qc~\x03\x07\xcd\xcbLg/)\x8b\x8b\xc6\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),
+    ("", "\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", None),
+    ("\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", "\x01", None),
+    ("\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p", "12345", None),
+    (repeat('\x01', 33), repeat('\x01', 32), None),
+  ]
+
+test "binary trie branch node encode":
+  for c in branchData:
+    let left   = b(c[0])
+    let right  = b(c[1])
+    let output = b(c[2])
+
+    try:
+      check output == encodeBranchNode(left, right)
+    except AssertionError as E:
+      check (E.msg == "len(leftChildNodeHash) == 32 ") or (E.msg == "len(rightChildNodeHash) == 32 ")
+    except:
+      doAssert(false)
+
+const
+  leafData = [
+    ("\x03\x04\x05", "\x02\x03\x04\x05"),
+    ("", None)
+  ]
 
```

Comment:
Let's add a randomized test similar to this one:
https://github.com/ethereum/py-trie/blob/master/tests/test_bin_trie.py#L20

You can use the Nim's quickcheck library for this:
https://github.com/alehander42/nim-quicktest